### DISCUSSION TITLE
The double type stored in the JSONB can only hold 15 decimal places

### DISCUSSION THREAD

**Discussion by jockerfan** [apache/doris#31327] [A - General / Q&A]
JSONB类型字段中存储的double类型字段只能保存小数点后15位

原始数据                0.5126184099180511
JSONB中的数据     0.512618409918051
DOBLE格式的数据 0.51261840991805108
存储的数据不一致是什么原因呢？

The double type fields stored in JSONB type columns can only preserve up to 15 decimal places.

Original Data: 0.5126184099180511
Data in JSONB: 0.512618409918051
Data in double format: 0.51261840991805108

What could be the reason for the inconsistency in the stored data?

**Comment by LemonLiTree** [comment]
一般情况下 double就是小数点后15位，但在64位操作系统上可以达到 17位，
如果只是double类型 就比较灵活，但是在jsonb里会通过sizeof(double) 来控制数据，尾数固定只有52位，所以只有后15位

**Reply by jockerfan** [reply]
谢谢
